rands.c

```diff
@@ -19,38 +20,30 @@ long _randseed = 0L;
 
 /*****************************************************/
 /**
-  \fn void RandSeed(signed long seed)
-  \brief Resets the random number seed.
+  \brief Sets the random number seed. This should only be done ONCE per iteration.
 
- The seed is set to negative when this routine
- is called, so the generator routines
- ( eg, RandUni()) can tell that it has
- changed.  If called with seed==0,
- _randseed is reset from process time.
- '% 0xffff' is due to a bug in RandUni()
- that conks if seed is too large; should
- be removed in the near future.
+  \param Seed is the initial state of the system. 0 indicates a random seed.
+  \param pcg_rng is the random number generator to set.
 
- \param seed The seed.
-
- cwb - 6/27/00
 */
-void RandSeed(signed long seed) {
+void RandSeed(signed long seed, pcg32_random_t* pcg_rng) {
 
-	if (seed == 0L) {
-		_randseed = ((long) time(NULL ));
-		if (_randseed == -1) {
-      sw_error(-1, "ERROR: RandSeed(0) called, but time() not available\n");
-		}
-		/*    _randseed %= 0xffff; */
-		_randseed *= -1;
-	} else {
-		_randseed = labs(seed) * -1;
-	}
+  if(seed == 0){
+    //seed with a random value. Uses the system time to generate
+    //a pseudo-random seed.
+    pcg32_srandom_r(pcg_rng, time(NULL), stream);
+  }
+  else {
+    //Seed with a specific value.
+    pcg32_srandom_r(pcg_rng, (int) seed, stream);
+  }
+
+  //Increment the stream so no two generators have the same sequence.
+  stream++;
 
 #ifndef RSOILWAT
 #if RAND_FAST

```

Comment:
Please remove `RAND_FAST` -- we eliminated `RandUni_fast` and `RandUni_good` and thus no longer need to select between them.